stickynotes/stickynotes.c

```diff
@@ -50,978 +50,1055 @@ set_icon_geometry (GdkWindow *window,
-			if (note->color)
-				xmlNewProp (node, XML_CHAR ("color"), XML_CHAR (note->color));
-			if (note->font_color)
-				xmlNewProp (node, XML_CHAR ("font_color"),
-						XML_CHAR (note->font_color));
-			if (note->font)
-				xmlNewProp (node, XML_CHAR ("font"), XML_CHAR (note->font));
-			if (note->locked)
-				xmlNewProp (node, XML_CHAR ("locked"), XML_CHAR ("true"));
-			xmlNewProp (node, XML_CHAR ("x"), XML_CHAR (x_str));
-			xmlNewProp (node, XML_CHAR ("y"), XML_CHAR (y_str));
-			xmlNewProp (node, XML_CHAR ("w"), XML_CHAR (w_str));
-			xmlNewProp (node, XML_CHAR ("h"), XML_CHAR (h_str));
-			if (note->workspace > 0)
-			{
-				char *workspace_str;
-
-				workspace_str = g_strdup_printf ("%i",
-						note->workspace);
-				xmlNewProp (node, XML_CHAR ("workspace"), XML_CHAR (workspace_str));
-				g_free (workspace_str);
-			}
-		}
-
-		/* Now that it has been saved, reset the modified flag */
-		gtk_text_buffer_set_modified(buffer, FALSE);
-
-		g_free(x_str);
-		g_free(y_str);
-		g_free(w_str);
-		g_free(h_str);
-		g_free(body);
-	}
-
-	/* The XML file is $HOME/.config/mate/stickynotes-applet, most probably */
-	{
-		gchar* path = g_build_filename(g_get_user_config_dir(), "mate", NULL);
-		gchar* file = g_build_filename(path, "stickynotes-applet.xml", NULL);
-		g_mkdir_with_parents(path, S_IRWXU);
-		g_free(path);
-
-		xmlSaveFormatFile(file, doc, 1);
-
-		g_free(file);
-	}
-
-	xmlFreeDoc(doc);
-
-	save_scheduled = FALSE;
-
-	return FALSE;
+    WnckScreen *wnck_screen;
+    const gchar *title;
+    GtkTextBuffer *buffer;
+    GtkTextIter start, end;
+    gchar *body;
+
+    gint i;
+
+    /* Create a new XML document */
+    xmlDocPtr doc = xmlNewDoc (XML_CHAR ("1.0"));
+    xmlNodePtr root = xmlNewDocNode (doc, NULL, XML_CHAR ("stickynotes"), NULL);
+
+    xmlDocSetRootElement (doc, root);
+    xmlNewProp (root, XML_CHAR ("version"), XML_CHAR (VERSION));
+
+    wnck_screen = wnck_screen_get_default ();
+    wnck_screen_force_update (wnck_screen);
+
+    /* For all sticky notes */
+    for (i = 0; i < g_list_length (stickynotes->notes); i++) {
+        WnckWindow *wnck_win;
+        gulong xid = 0;
+
+        /* Access the current note in the list */
+        StickyNote *note = g_list_nth_data (stickynotes->notes, i);
+
+        /* Retrieve the window size of the note */
+        gchar *w_str = g_strdup_printf ("%d", note->w);
+        gchar *h_str = g_strdup_printf ("%d", note->h);
+
+        /* Retrieve the window position of the note */
+        gchar *x_str = g_strdup_printf ("%d", note->x);
+        gchar *y_str = g_strdup_printf ("%d", note->y);
+
+        xid = GDK_WINDOW_XID (gtk_widget_get_window (note->w_window));
+        wnck_win = wnck_window_get (xid);
+
+        if (!g_settings_get_boolean (stickynotes->settings, "sticky") &&
+            wnck_win)
+            note->workspace = 1 +
+                wnck_workspace_get_number (
+                wnck_window_get_workspace (wnck_win));
```

Comment:
`wnck_window_get_workspace` should not be aligned with `wnck_workspace_get_number`